### ISSUE TITLE
cuda out of memory

### ISSUE THREAD

**Issue by Ansa-Saeed** [ghdtjr/A-LLMRec#3]
when i run the code at 48 gpu so face the error out of memory. i run the stage 1 only 1 epochs its run at 48 gpu. BUt now when i run the stage 2 so face the issue Cuda out of memory. 
![image](https://github.com/user-attachments/assets/e4d7642a-d6b0-44ec-a819-09f129ef24dd)
![image](https://github.com/user-attachments/assets/a602a8c6-f89b-4f40-a7fa-8a1baf5c8aa2)
![image](https://github.com/user-attachments/assets/c56477b9-d1d4-415f-a6ef-828f3f234427)
kindly suggest me how to avoid the error

**Comment by haorongchen1015** [comment]
Hi Ansa,

I hope this message finds you well. I’m reaching out because I encountered an issue while setting up the environment this project. Specifically, I’m seeing the following error in the requirements.txt file:
"""
ERROR: Invalid requirement: '_libgcc_mutex=0.1=main' (from line 4 of requirements.txt)
Hint: = is not a valid operator. Did you mean == ?
"""
Have you encountered this issue before? If so, could you please share how you resolved it? Any advice or guidance would be greatly appreciated.

Thank you in advance for your help!

Best regards,
Haorong

**Comment by Ansa-Saeed** [comment]
Try using _libgcc_mutex==0.1, but in my case, it didn't work. I am installing only specific libraries separately, like:

    transformers==4.32.1
    sentencetransformers==2.2.2
    accelerate==0.25.0
    pytz
    torch==2.1.2

Please see the 
[requirements.txt](https://github.com/user-attachments/files/16400709/requirements.txt)
 file. Not all libraries were installed, so I am installing only the specific ones and running the code on Jupyter Notebook.

**Comment by haorongchen1015** [comment]
I see. 
Many thanks

**Comment by ghdtjr** [comment]
Which dataset are you planning to use for learning the model?

The length of each item title directly impacts memory usage. Therefore, based on the specific dataset you might decrease the batch size or limit the item title length.

**Comment by Ansa-Saeed** [comment]
Can you please tell me about num of user and no of item which version of amazon dataset you used *?

In your paper you used 2014 version of amazon dataset i run the code on beauty dataset but when i run the preprocess file so i got user num and item num 82331 70286  with threshold 4 [cid:fd827280-02dd-4bff-8836-24f331ce1d99]
But in your paper beauty result other  how its possible kindly guide me Thanks ![cid:193f6983-306e-4575-87f7-f7edb8aa7e65]
________________________________

**Comment by ghdtjr** [comment]
We used the 2018 Amazon dataset the number of users and items after the preprocessing are described in our paper.

Not only the threshold with the number of interactions, we preprocessed the raw data with the rating. If you considered only the interaction threshold, there might be some differences with our experiment result.

However, I suggest running the code with batch size 1 if you still face an out-of-memory issue.
